src/adm/tiles.rs

```diff
@@ -211,13 +211,22 @@ pub async fn get_tiles(
         metrics.incr_with_tags("filter.adm.empty_response", Some(tags));
     }
 
+    // update the filters if needed.
+    if settings.adm_live_update && state.filter.read().unwrap().requires_update().await? {
```

Comment:
This should be needed generally with the background updater.

I think the one time it might be needed is during startup if the background updater hasn't completed reading the bucket yet. Instead Could `Server::with_settings` call `AdmFilter::update`, blocking on starting the server until it's finished?

Reply:
I was having problems with actix calling async calls from inside sync calls which lead to this. Basically, things compile, but then the app fails because handler is already in use. I tried several different solutions to get around that, but nothing worked.

Reply:
Most of the startup is within the async runtime so this looks doable, here's what I mean: https://github.com/mozilla-services/contile/compare/feat/324-bucket...feat/324-bucket-block-init

Though that branch will crash contile at startup if the bucket reading fails for whatever reason. I'm not sure if that's worse than ignoring it and serving empty tiles, though

Reply:
Yeah, I was trying not to do that, partly because the code got ugly kind of fast. Although I might have a fix for that...